src/features/patients/hooks/usePatientLookup.ts

```diff
@@ -0,0 +1,306 @@
+    case ParameterTypes.date:
+      return (key: string, date: Date) =>
+        isValid(date) ? `${key}=${format(date, "ddMMyyyy")}` : "";
+    case ParameterTypes.number:
+      return (key: string, number: number) => `${key}=${Number(number)}`;
+    default:
+      return () => "";
+  }
+};
+
+const getQueryString = (params: PatientSearchParameterType[]) => {
+  const query = params
+    .filter((param) => param.value)
+    .map((param) => {
+      const { key, value, type } = param;
+      const formatter = getFormatter(type);
+      return formatter(key, value);
+    });
+  return query.join("&");
+};
+
+const getPatientQueryString = ({
+  firstName = "",
+  lastName = "",
+  dateOfBirth = "",
+  limit = null,
+  offset = null,
+} = {}) => {
+  const queryParams: PatientSearchParameterType[] = [
+    {
+      key: PARAMETERS.firstName.key,
+      value: firstName,
+      type: PARAMETERS.firstName.type,
+    },
+    {
+      key: PARAMETERS.lastName.key,
+      value: lastName,
+      type: PARAMETERS.lastName.type,
+    },
+    {
+      key: PARAMETERS.dateOfBirth.key,
+      value: dateOfBirth,
+      type: PARAMETERS.dateOfBirth.type,
+    },
+    { key: PARAMETERS.offset.key, value: offset, type: PARAMETERS.offset.type },
+    { key: PARAMETERS.limit.key, value: limit, type: PARAMETERS.limit.type },
+  ];
+  return getQueryString(queryParams);
+};
+
+export const getPatientRequestUrl = (params: any) => {
+  return `${getPatientUrl()}?${getPatientQueryString(params)}`;
+};
+
+/**
+ * Hook to help a component to be able to find patients through the patient lookup API.
+ */
+export const usePatientLookup = () => {
+  const [
+    { data, loading, searchedWithNoResults, error, limit = BATCH_SIZE, noMore },
+    dispatch,
+  ] = useReducer(reducer, [], initialState);
+
+  const searchOnline = (searchParams: SearchParameters) => {
+    const paramsWithLimits = { ...searchParams, limit };
+
+    dispatch(LookupAction.clear());
+    dispatch(LookupAction.start());
+    axios
+      .get(getPatientRequestUrl(paramsWithLimits), {
+        withCredentials: true,
+      })
+      .then((data) => {
+        const action = data.data.length
+          ? LookupAction.success(data)
+          : LookupAction.noResult();
+        dispatch(action);
+      })
+      .catch((error) => dispatch(LookupAction.error(error)));
+  };
+
+  const searchMore = (searchParams: SearchParameters) => {
+    const paramsWithOffset = { ...searchParams, limit, offset: data.length };
+
+    dispatch(LookupAction.gettingMore());
+    axios
+      .get(getPatientRequestUrl(paramsWithOffset), {
+        withCredentials: true,
+      })
+      .then((data) => dispatch(LookupAction.addMore(data)))
+      .catch((error) => dispatch(LookupAction.error(error)));
+  };
+
+  const refresh = () => dispatch(LookupAction.clear());
+
+  return {
+    refresh,
+    data,
+    error,
+    noMore,
+    loading,
+    searchedWithNoResults,
+    searchOnline,
+    searchMore,
+  };
+};
```

Comment:
Nice, although it's arguably a little easier utilising react-query : https://react-query.tanstack.com/guides/paginated-queries

Reply:
I was trying to avoid another package. I regert that now.

Reply:
We're already using it hehe

Reply:
🤦

Reply:
perhaps a new issue though. this one is dragging on :)